temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -1585,6 +1585,22 @@ private void processContinueAsNewWorkflowExecution(
         parentChildInitiatedEventId);
   }
 
+  private void processWorkflowCompletionCallbacks(RequestContext ctx) {
+    Optional<HistoryEvent> completionEvent = getCompletionEvent(ctx.getEvents());
+    if (!completionEvent.isPresent()) {
+      return;
+    }
+
+    for (Callback cb : startRequest.getCompletionCallbacksList()) {
+      if (!cb.hasNexus()) {
+        // test server only supports nexus callbacks currently
+        log.warn("skipping non-nexus completion callback");
```

Comment:
This should just be an invalid argument instead of this log that will get lost and leave users confused.

Reply:
I added validation on the start request to throw invalid argument if there are any non-nexus callbacks. I like leaving this as a warning in case someone still somehow gets a non-nexus callback in so that we don't fail the workflow.